pkg/controllers/selection/controller.go

```diff
@@ -114,6 +117,21 @@ func (c *Controller) selectProvisioner(ctx context.Context, pod *v1.Pod) (errs e
 	return nil
 }
 
+func getRelaxOptions(provisioners []*provisioning.Provisioner) RelaxOptions {
```

Comment:
This breaks the encapsulation of `preferences.go`. What can do to maintain that encapsulation?

Reply:
Sorry, not following this comment. RelaxOptions is declared in preferences.go and I introduced it to avoid having preferences know about provisioners.